packages/create-llama/templates/components/vectordbs/python/pg/utils.py

```diff
@@ -0,0 +1,68 @@
+import os
+from llama_index.vector_stores import PGVectorStore
+from pydantic import BaseModel
+from urllib.parse import urlparse
+
+
+class PGConnectionElements(BaseModel):
+    user: str
+    password: str
+    host: str
+    port: int
+    database: str
+
+
+def destruct_pg_connection_string(conn_string: str) -> PGConnectionElements:
+    """
+    Destructs the PostgreSQL connection string into its elements.
+    """
+    parsed_url = urlparse(conn_string)
+
+    user = parsed_url.username
+    password = parsed_url.password
+    host = parsed_url.hostname
+    port = parsed_url.port
+    database = parsed_url.path[1:]
+
+    return PGConnectionElements(
+        user=user,
+        password=password,
+        host=host,
+        port=port,
+        database=database
+    )
+
+
+def get_pg_conn_elements_from_env() -> PGConnectionElements:
+    """
+    Returns the PostgreSQL connection elements from the environment variables:
+    [PGUSER, PGPASSWORD, PGHOST, PGPORT, PGDATABASE]
+    """
+    env_conn_str = os.environ.get("PG_CONNECTION_STRING")
+    if env_conn_str is not None:
+        return destruct_pg_connection_string(os.environ["PG_CONNECTION_STRING"])
+    else:
+        user = os.environ.get("PGUSER")
+        password = os.environ.get("PGPASSWORD")
+        host = os.environ.get("PGHOST")
+        port = int(os.environ.get("PGPORT", 5432))
+        database = os.environ.get("PGDATABASE")
+        return PGConnectionElements(
+            user=user,
+            password=password,
+            host=host,
+            port=port,
+            database=database
+        )
```

Comment:
It's a great idea to provide an alternative way to configure the DB parameters. 
Here I wouldn't do it, though as the generator is only generating `PG_CONNECTION_STRING` - if it doesn't exist, I would just throw an exception and tell the user to configure `PG_CONNECTION_STRING`

Reply:
as this method becomes then i one-liner, you can inline it to `init_pg_vector_store_from_env`

Reply:
> It's a great idea to provide an alternative way to configure the DB parameters. Here I wouldn't do it, though as the generator is only generating `PG_CONNECTION_STRING` - if it doesn't exist, I would just throw an exception and tell the user to configure `PG_CONNECTION_STRING`

this document says that the user can also set the credential variables (https://github.com/run-llama/LlamaIndexTS/blob/85da7985388d92cb71f60f34e7d205fa142a58e3/examples/pg-vector-store/README.md?plain=1#L22), is it related to this template or just setup the environment itself?

If so, because the PGVectorStore in llama_index required the async connection string so how about constructing the async one from the PG_CONNECTION_STRING by replacing the schema then passing them to PGVectorStore directly? (https://github.com/run-llama/llama_index/blob/2edd31c83b3648fc81eff636cebc183697958d12/llama_index/vector_stores/postgres.py#L130)

Reply:
I pushed an update, please help me to review it again, thank you!

Reply:
@leehuwuj that looks great thanks. The doc you're referring to is just an example for LlamaIndexTS; it's not about how to use the Python version